Sources/ApodiniDatabase/Query/QueryBuilder.swift

```diff
@@ -0,0 +1,124 @@
+        self.type = type
+//        self.queryString = queryString
+        self.parameters = extract(from: queryString)
+    }
+    
+    init(type: Model.Type, parameters: [FieldKey: String]) {
+        self.type = type
+        self.parameters = parameters
+    }
+    
+    private func extract(from queryString: String) -> [FieldKey: String] {
+        var foundParameters: [FieldKey: String] = [:]
+        let extractedQueryString = queryString.split(separator: "?")[1]
+        let queryParts = extractedQueryString.split(separator: "&").map { String($0) }
+        for part in queryParts {
+            let queryParameters = part.split(separator: "=").map { String($0) }
+            guard queryParameters.count == 2 else { fatalError("invalid query") }
+            let key = queryParameters[0]
+            let value = queryParameters[1]
+            guard let fieldKey = fieldKeys.first(where: { $0.description == key }) else {
+                continue
+            }
+            foundParameters[fieldKey] = value
+        }
+        return foundParameters
+    }
+    
+    internal func execute(on database: Fluent.Database) -> EventLoopFuture<[Model]> {
+        func accumulateFilters(queryBuilder: inout Fluent.QueryBuilder<Model>, key: FieldKey, value: String, method: DatabaseQuery.Filter.Method) {
+            if let intValue = Int(value) {
+                queryBuilder.filter(key, method, intValue)
+            } else if let boolValue = Bool(value) {
+                queryBuilder.filter(key, method, boolValue)
+            } else {
+                queryBuilder.filter(key, method, value)
+            }
+        }
+        
+        var queryBuilder = Model.query(on: database)
+        for (key, value) in parameters {
+            accumulateFilters(queryBuilder: &queryBuilder, key: key, value: value, method: .equal)
+        }
+        return queryBuilder.all()
+    }
+    
+    @discardableResult
+    internal static func fieldKeys<Model: DatabaseModel>(for type: Model.Type) -> [FieldKey] {
+        type.keys
+    }
+    
+    internal static func info(for type: Model.Type) -> [ModelInfo] {
+        var modelInfo: [ModelInfo] = []
+        do {
+            let keys = type.keys
+            let info = try typeInfo(of: type)
+            guard info.properties.count == type.keys.count else {
+                fatalError("wrong model")
+            }
+            for (index, propertyInfo) in info.properties.enumerated() {
+                if propertyInfo.name.replacingOccurrences(of: "_", with: "") == keys[index].description {
+                    let key = keys[index]
+                    let type = Self.fieldType(for: propertyInfo.type)
+                    modelInfo.append(ModelInfo(key: key, type: type))
+                }
+            }
+        } catch {
+            fatalError("failed to infer info")
+        }
+        return modelInfo
+    }
+    
+    // swiftlint:disable:next todo
+    //TODO: Find a better way to do this
```

Comment:
I would suggest to implement some small visitor pattern with an extension to `FieldProperty` that would allow you to access the type and the corresponding information using type checks instead of string representations, let me know if you need some more input on this 👍

Reply:
Its definitely a good idea to move away from checking string representations! 👍 Would your approach also allow to have some type info already on compile time? To have a real benefit, we would need to have the type info at compile time such that `Read` can initialise its `@Properties` with `Parameter<Type` and stop using `Parameter<String>` for everything

Reply:
I agree it would be really beneficial to have proper type-info here! While it does not make that much of a difference for HTTP url parameters, it does for other exporters which rely on JSON or protobuf. Also, it would look a bit ugly in the OpenAPI docs.

Reply:
@hendesi Yes, I could image that a visitor pattern based approach would be able to address this issue and would be able to get the type info on compile time. @nerdsupremacist or I can take a look at this in the work meeting and provide some input on how to implement this 🚀

Reply:
@PSchmiedmayer Thanks! I already implemented the visitor pattern on my local branch and it works quite well. Might have some additional questions which we could discuss in the work meeting 👍